framework/i18n/Formatter.php

```diff
@@ -787,6 +787,53 @@ public function asRelativeTime($value, $referenceTime = null)
         }
     }
 
+    /**
+     * Formats the value as a time interval.
+     * @param \DateInterval|string|integer $value the value to be formatted, in ISO8601 duration format or number of seconds.
+     * @return string the formatted result.
+     */
+    public function asDuration($value)
+    {
+        if ($value === null) {
+            return $this->nullDisplay;
+        }
+        if ($value instanceof \DateInterval) {
+            $negative = $value->invert;
+            $interval = $value;
+        } elseif (is_numeric($value)) {
+            $negative = $value < 0;
+            $zeroDateTime = (new DateTime())->setTimestamp(0);
+            $valueDateTime = (new DateTime())->setTimestamp(abs($value));
+            $interval = $valueDateTime->diff($zeroDateTime);
+        } elseif (strpos($value, '-') !== false) {
```

Comment:
No, it doesn't. Previous conditions checks if this is a digit. This condition checks if this is a ISO duration format starting with a dash.